### PR TITLE
Separate the DMAPI version from Interop Version

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl:
Fixed `-public` always being set in DreamDaemon command lines. Sometimes twice, other times overriding the intended `-invisible` setting during deployments.
/:cl:

:cl: DMAPI
The DMAPI is now versioned separately from the TGS interop APIs it provides.
Removed `/datum/tgs_revision_information/test_merge/var/time_merged`.
Renamed `/datum/tgs_revision_information/test_merge/var/pull_request_commit` to `head_commit`.
/:cl:

:cl: HTTP API
The `dmApiVersion` field in the ServerInformation and CompileJob models refers to the server's interop version.
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
```
  X TestServer [2m 34s]
  Error Message:
   Assert.Fail failed. DreamDaemon didn't shutdown within the timeout!
```

File: src/DMAPI/tgs/v5/api.dm [hunk 1]
```diff
@@ -18,7 +18,9 @@
 	var/initialized = FALSE
 
 /datum/tgs_api/v5/ApiVersion()
-	return new /datum/tgs_version(TGS_DMAPI_VERSION)
+	return new /datum/tgs_version(
+		#include "interop_version.dm"
```
Comment (AffectedArc07) [inline src/DMAPI/tgs/v5/api.dm:22]: I hate this

File: tests/Tgstation.Server.Tests/VersionsTest.cs [hunk 2]
```diff
@@ -99,7 +99,14 @@ public void TestDmapiVersion()
 
 			Assert.IsTrue(Version.TryParse(versionLine, out var actual));
 			Assert.AreEqual(expected, actual);
-			Assert.AreEqual(expected, DMApiConstants.Version);
+		}
+
+		public void TestInteropVersion()
```
Comment (AffectedArc07) [inline tests/Tgstation.Server.Tests/VersionsTest.cs:104]: Doesnt this need `[TestMethod]` above it?

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: tests/Tgstation.Server.Tests/VersionsTest.cs (hunk 2 above)
Comment (Cyberboss) [reply tests/Tgstation.Server.Tests/VersionsTest.cs:104]: FUCK

**Comment by Cyberboss** [comment]
Getting annoyed

**Pull request merged**